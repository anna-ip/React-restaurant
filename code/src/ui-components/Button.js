import React from "react";
import styles from "./Button.module.scss";

const Button = ({ className, value, onClick }) => {
  return (
    <>
      <button
        className={className + " " + styles.btn}
        onClick={onClick}
        value={value}
      >
        {value}
      </button>
    </>
  );
};

export default Button;
