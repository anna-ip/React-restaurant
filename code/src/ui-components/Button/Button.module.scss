@import '../../styles/_variables.scss';

.btn {
  cursor: pointer;
  border: none;
  background: inherit;
  color: $red-color;

  &:focus,
  &:hover {
    transform: scale(1.1);
  }
}
