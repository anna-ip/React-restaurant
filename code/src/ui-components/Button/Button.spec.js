import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

describe("<Button/>", () => {
  test("renders", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.exists()).toBe(true);
  });

  test("snapshot", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });
});
