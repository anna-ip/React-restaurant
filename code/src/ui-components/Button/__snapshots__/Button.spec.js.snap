// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> snapshot 1`] = `
<Fragment>
  <button
    className="undefined btn"
  />
</Fragment>
`;
